src/smtencoding/FStar.SMTEncoding.Encode.fs

```diff
@@ -742,16 +778,30 @@ and encode_term (t:typ) (env:env_t) : (term         (* encoding of t, expects t
             | Inr (eff, _) -> TypeChecker.Util.is_pure_or_ghost_effect env.tcenv eff |> not
           in
 
+          let reify_comp_and_body env c body =
+            let reified_body = reify_body env.tcenv body in
+            let c = match c with
+              | Inl lc ->
+                let typ = FStar.TypeChecker.Util.reify_comp ({env.tcenv with lax=true}) lc U_unknown in
+                Inl (U.lcomp_of_comp (S.mk_Total typ))
+
+              (* In this case we don't have enough information to reconstruct the *)
+              (* whole computation type and reify it *)
+              | Inr (eff_name, _) -> c
```

Comment:
might this allow us to mistakenly reify computations with non-trivial pre-conditions?